resources/resources.go

```diff
@@ -53,22 +53,34 @@ func EmbeddedDirExists(path string) (bool, error) {
 }
 
 // FetchHTTP
-// Fetch a resource from a remote HTTP server.
-func FetchHTTP(uri string, rsrc string) (io.ReadCloser, error) {
-	resp, remoteErr := http.Get(uri + "/" + rsrc)
+// Fetch a resource from a remote HTTP server with bearer token auth.
+func FetchHTTP(uri string, rsrc string, auth string) (io.ReadCloser, error) {
+	req, reqErr := http.NewRequest("GET", uri+"/"+rsrc, nil)
+	if reqErr != nil {
+		return nil, reqErr
+	}
+	req.Header.Add("authorization", "Bearer "+auth)
+	resp, remoteErr := http.DefaultClient.Do(req)
 	if remoteErr != nil {
 		return nil, remoteErr
-	} else if resp.StatusCode != 200 {
+	}
+	if resp.StatusCode != 200 {
 		return nil, errors.New(fmt.Sprintf("HTTP status code %d",
 			resp.StatusCode))
 	}
 	return resp.Body, nil
 }
 
+
 // SizeHTTP
-// Get the size of a resource from a remote HTTP server.
-func SizeHTTP(uri string, rsrc string) (uint, error) {
-	resp, remoteErr := http.Head(uri + "/" + rsrc)
+// Get the size of a resource from a remote HTTP server with bearer token auth.
+func SizeHTTP(uri string, rsrc string, auth string) (uint, error) {
+	req, reqErr := http.NewRequest("HEAD", uri+"/"+rsrc, nil)
+	if reqErr != nil {
+		return 0, reqErr
+	}
+	req.Header.Add("authorization", "Bearer "+auth)
```

Comment:
Does this work if we have no token provided and auth is not required?

Reply:
Yep, the HF API ignores the authorization header if the repository is public.

Reply:
I'd gate it on if `auth` is empty or not, for neatness sake.